src/estimagic/inference/bootstrap_samples.py

```diff
@@ -34,6 +45,29 @@ def get_bootstrap_indices(data, rng, cluster_by=None, n_draws=1000):
     return bootstrap_indices
 
 
+def _get_probs_for_bootstrap_indices(data, weight_by, cluster_by):
```

Comment:
I prefer the function name `_calculate_bootstrap_indices_weights`. In that case, the first line of the docstring needs to be updated as well.